### PR TITLE
feat: allocation % for both ticker and industry

### LINKED ISSUE #30: API endpoint for get portfolio allocation based on asset ticker % within portfolio
## Features

Endpoint result should look like this:
```json
[
  {
    "assetTicker": "AAPL",
    "percentage": 0.35
  }
]
```

### LINKED ISSUE #41: [BUG] Request for a non-existent portfolio pid returns success 200 with empty array
## Context

Request for a non-existent portfolio (eg value of `-1`) returns success 200 with empty array

```json
{
  "portfolioAssetList": []
}
```

## To Reproduce

1. checkout `main` & setup repo
2. visit `http://localhost:8080/api/portfolio/assets/-1`

   OR 

   ```sh
   curl -X 'GET' \
     'http://localhost:8080/api/portfolio/assets/-1' \
     -H 'accept: application/json'
   ```

## Expected behavior

Return 404, Not Found error

## Screenshots

screenshot

## Priority

Low

### PR DESCRIPTION
**Pull request by TimothyNgoi98**
# Get allocation % by ticker/industry + bug fix
closes #30 closes #41 
had to refactor code for get allocation by industry due to removal of asset id which has since changed to asset ticker

## Prerequisite steps
Start up the Springboot application 
```
make.bat dev
```

## Get allocation % by ticker
Send a HTTP GET request to the following endpoint http://localhost:8080/api/portfolio/{pid}/allocation/ticker
```json
[
    {
        "assetTicker": "NFLX",
        "percentage": 0.12162162
    },
    {
        "assetTicker": "GOOGL",
        "percentage": 0.067567565
    },
    {
        "assetTicker": "NVDA",
        "percentage": 0.10810811
    },
    {
        "assetTicker": "AAPL",
        "percentage": 0.4054054
    },
    {
        "assetTicker": "TSLA",
        "percentage": 0.2027027
    },
    {
        "assetTicker": "AMZN",
        "percentage": 0.0945946
    }
]
```

## Get allocation % by industry
Send a HTTP GET request to the following endpoint http://localhost:8080/api/portfolio/{pid}/allocation/industry
```json
[
    {
        "percentage": 0.12162162,
        "industry": "Entertainment"
    },
    {
        "percentage": 0.8783784,
        "industry": "Technology"
    }
]
```

### CODE REVIEW

File: src/main/java/com/backend/controller/PortfolioController.java [hunk 1]
```diff
@@ -219,47 +225,90 @@ public List<Map<String, Object>> getTransactionsByPortfolioId(@PathVariable int
...
 
+	@GetMapping(path = "/portfolio/{pid}/allocation/ticker")
+	public List<Map<String, Object>> getAllocationPercentageByTicker(@PathVariable long pid) {
+		Portfolio portfolio = portfolioService.findByPid(pid);
+		if (portfolio == null) {
+			throw new PortfolioNotFoundException(pid);
+		}
+
+		List<PortfolioAsset> portfolioAssetList = portfolioAssetService.findAllByPortfolioId(pid);
+		if (portfolioAssetList == null) {
```
Comment (DavidLHW) [inline src/main/java/com/backend/controller/PortfolioController.java:236]: should it be `portfolioAssetList.isEmpty()` instead?

File: src/main/java/com/backend/controller/PortfolioController.java [hunk 2]
```diff
@@ -219,47 +225,90 @@ public List<Map<String, Object>> getTransactionsByPortfolioId(@PathVariable int
...
 		Portfolio portfolio = portfolioService.findByPid(pid);
 		if (portfolio == null) {
-			throw new PortfolioAssetNotFoundException(pid);
+			throw new PortfolioNotFoundException(pid);
 		}
 
-		Map<String, Integer> industryMap = new HashMap<String, Integer>();
 		List<PortfolioAsset> portfolioAssetList = portfolioAssetService.findAllByPortfolioId(pid);
-		int totalSize = portfolioAssetList.size();
+		if (portfolioAssetList == null) {
```
Comment (DavidLHW) [inline src/main/java/com/backend/controller/PortfolioController.java:276]: same as abv:

```java
if (portfolioAssetList.isEmpty()) {
  // ...
}
```

**Review by DavidLHW — APPROVED** [review]
Thanks for the PR! Some minor comments, else good to merge :)

File: src/main/java/com/backend/controller/PortfolioController.java (hunk 2 above)
Comment (TimothyNgoi98) [reply src/main/java/com/backend/controller/PortfolioController.java]: ah thanks for pointing out!

**Pull request merged**